File: plop-templates/adapter/acquiring/src/client/RemoteClientImpl.java
package com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.*;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

{{#if_eq doc true}}
/**
 * Здесь находится реализация методов вызова 3ей стороны.
 */
{{/if_eq}}
@Slf4j
@Service
@RequiredArgsConstructor
public class RemoteClientImpl implements RemoteClient {

    private final RestTemplate restTemplate;
    private final ObjectMapper objectMapper;
{{#if_eq doc true}}
//    todo implement and use
//    private final ResponseConverter<BaseResponse> responseConverter;
{{/if_eq}}

    @Value("${adapter.url}")
    private String basePath;

    @Override
    public PreAuthResponse preAuth(PreAuthRequest request) {
        return null;
    }

    @Override
    public AuthResponse auth(AuthRequest request) {
        return null;
    }

    @Override
    public CaptureResponse capture(CaptureRequest request) {
        return null;
    }

    @Override
    public CancelResponse cancel(CancelRequest request) {
        return null;
    }

    @Override
    public RecurrentResponse recurrent(RecurrentRequest request) {
        return null;
    }

    @Override
    public StatusResponse status(StatusRequest request) {
        return null;
    }

}

File: plop-templates/adapter/acquiring/src/client/RemoteClientImpl.kt
package com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client

import com.fasterxml.jackson.databind.ObjectMapper
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.AuthRequest
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.AuthResponse
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.CancelRequest
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.CancelResponse
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.CaptureRequest
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.CaptureResponse
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.PreAuthRequest
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.PreAuthResponse
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.RecurrentRequest
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.RecurrentResponse
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.StatusRequest
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.StatusResponse
import mu.KotlinLogging
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.client.RestTemplate

{{#if_eq doc true}}
/**
 * Здесь находится реализация методов вызова 3ей стороны.
 */
{{/if_eq}}
@Service
class RemoteClientImpl(
    private val restTemplate: RestTemplate,
    private val objectMapper: ObjectMapper,
{{#if_eq doc true}}
//        todo implement and use
//        private val responseConverter: ResponseConverter<BaseResponse>,
{{/if_eq}}
    @Value("\${adapter.url}") private val basePath: String
) : RemoteClient {

    private val log = KotlinLogging.logger {}

    override fun preAuth(request: PreAuthRequest): PreAuthResponse {
        TODO("Not yet implemented")
    }

    override fun auth(request: AuthRequest): AuthResponse {
        TODO("Not yet implemented")
    }

    override fun capture(request: CaptureRequest): CaptureResponse {
        TODO("Not yet implemented")
    }

    override fun cancel(request: CancelRequest): CancelResponse {
        TODO("Not yet implemented")
    }

    override fun recurrent(request: RecurrentRequest): RecurrentResponse {
        TODO("Not yet implemented")
    }

    override fun status(request: StatusRequest): StatusResponse {
        TODO("Not yet implemented")
    }
}
